### PR TITLE
Add aten::_pin_memory, aten::is_pinned

### PR DESCRIPTION
**Pull request by hjhee**
- _pin_memory
- is_pinned

### CODE REVIEW

**Comment by fengyuan14** [comment]
Please add an example test case, since we haven't got a case to cover the operator.

File: src/aten/PinnedMemoryAllocator.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#include <ATen/ATen.h>
+#include <ATen/CPUFunctions.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/XPUNativeFunctions.h>
+#include <ATen/xpu/PinnedMemoryAllocator.h>
+#include <ATen/xpu/detail/XPUHooks.h>
```
Comment (fengyuan14) [inline src/aten/PinnedMemoryAllocator.cpp:6]: Is the header useful? or remove it?

**Review by dvrogozh — COMMENTED** [review]
This should fix https://github.com/pytorch/pytorch/issues/126491. Can you, please, add a link to commit message and PR to this issue?

**Comment by dvrogozh** [comment]
> This should fix https://github.com/pytorch/pytorch/issues/126491.

Tried it out. Indeed, this seems to fix the issue.

**Comment by fengyuan14** [comment]
No fail in pre-ci. Retrigger pre-ci.

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**